### DISCUSSION TITLE
backslash in KaTeX is wrong

### DISCUSSION THREAD

**Discussion by binLep** [imfing/hextra#368] [Q&A] [answered]
**Description**

Maybe the character "\\" is not escaped

In the example below, the input comes from obsidian editor, the output comes from hextra.

And "\\\\" cannot implement line break. it must be "\\\\\\\\" or "\\newline"

**Expected Behavior**

- Error input

![图片](https://github.com/imfing/hextra/assets/45154147/f26192c7-cb07-458d-9d5b-9eef707a3a46)

Result（Correct）

![图片](https://github.com/imfing/hextra/assets/45154147/c9f135c9-9b2e-43a7-aa7d-d3fd0803e07c)

- Correct input

![图片](https://github.com/imfing/hextra/assets/45154147/3d68e2e2-5e0d-49e8-8cb4-eef5c37bb92c)

Result（Error）

![图片](https://github.com/imfing/hextra/assets/45154147/4f41c3f5-cbb1-43ea-a061-2f44ad8eed2e)

**Actual Behavior**

- Correct input

![图片](https://github.com/imfing/hextra/assets/45154147/3d68e2e2-5e0d-49e8-8cb4-eef5c37bb92c)

Result（Correct）

![图片](https://github.com/imfing/hextra/assets/45154147/c9f135c9-9b2e-43a7-aa7d-d3fd0803e07c)

**Environment**

- Hugo Version: [v0.124.1]
- Browser/OS: [Edge]
- Theme Version: [v0.7.3]

**Comment by imfing** [comment]
Hi @binLep 
Could you please try enabling the passthrough in the Hugo config, as described in the documentation (https://gohugo.io/content-management/mathematics/#step-1)? this may help solve the issue

**Comment by binLep** [comment]
> Hi @binLep Could you please try enabling the passthrough in the Hugo config, as described in the documentation (https://gohugo.io/content-management/mathematics/#step-1)? this may help solve the issue

Thanks for your proposal, I have tried, but it seems to have no effect

Hugo's examples are similar to parenthesis-matching scenarios, like this

```javascript
displayMath: [['\\[', '\\]'], ['$$', '$$']],  // block
inlineMath: [['\\(', '\\)']]                  // inline
```

The scenario I propose is just a single symbol, including "\\&" and "\\\\".

🥲

**Comment by imfing** [comment, marked as answer]
@binLep with the `passthrough` enabled, it should work:

example `markup` setting from `hugo.yaml`:

```yaml
markup:
  goldmark:
    renderer:
      unsafe: true
    extensions:
      passthrough:
        delimiters:
          block:
          - - \[
            - \]
          - - $$
            - $$
          inline:
          - - \(
            - \)
        enable: true
```

result:

![image](https://github.com/imfing/hextra/assets/5097752/c0937678-c212-4360-af75-59faf5d3481a)


> Hugo's examples are similar to parenthesis-matching scenarios, like this

that's not quite true. the `passthrough` feature enables the math formula content between `$$` to be escaped from Hugo's Markdown engine, so you don't need to explicitly escape `\`.

**Reply by binLep** [reply]
I tried your plan. You are right. Thank you!
